docs/cli/stacks/nesting.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Nesting Stacks
+description: Learn how stacks help you to split up your infrastructure code and configuration such as Terraform into isolated units.
+---
+
+# Nesting Stacks
+
+## What are nested Stacks?
+
+Stacks in Terramate can be nested, allowing you to map your infrastructure code as a tree and execute commands
+hierarchically, leading to a natural organization of your infrastructure resources with Infrastructure as Code.
+
+::: tip
+Nesting stacks is a recommended best practice for grouping related infrastructure, such as environments or services that
+should be treated as a unit while still separating state with child stacks to profit from stacks-related benefits such as
+fast execution runtimes.
+:::
+
+A simple example would be launching an EC2 instance in AWS. The instance must be launched in a subnet which must belong
+to a vpc which in turn belongs to an account so the cloud hierarchy would look something like:
+
+<img src="../assets/nested-stacks.png" width="700px" alt="Terramate Nested Stacks Overview" />
```

Comment:
This diagram does not seem to fit the style of the page (because of the white background). I have two suggestions:

* Use a dark background image
* Rremove `require vpc.id` and `require subnet.id` from the diagram (because they seem unnecessary and not explained in the docs)
* We could use mermaid syntax to generate this diagram so its easier to update it in future (vitepress seems to have a mermaid plugin)


We could also use the mermaid syntax for it, see some examples below:

```
graph TD;
    A["vpc"]-->B["network_acl"];
    A-->C["internet_gateway"];
    A-->D["subnet"];
    A-->E["route_table"];
    A-->F["security_group"];
    D-->G["network_interface"];
    D-->H["ec2"];
```

```
graph LR;
    A["vpc"]-->B["network_acl"];
    A-->C["internet_gateway"];
    A-->D["subnet"];
    A-->E["route_table"];
    A-->F["security_group"];
    D-->G["network_interface"];
    D-->H["ec2"];
```


```mermaid
graph TD;
    A["vpc"]-->B["network_acl"];
    A-->C["internet_gateway"];
    A-->D["subnet"];
    A-->E["route_table"];
    A-->F["security_group"];
    D-->G["network_interface"];
    D-->H["ec2"];
```

```mermaid
graph LR;
    A["vpc"]-->B["network_acl"];
    A-->C["internet_gateway"];
    A-->D["subnet"];
    A-->E["route_table"];
    A-->F["security_group"];
    D-->G["network_interface"];
    D-->H["ec2"];
```

Reply:
awesome suggestion thank you!